client/src/Components/Pages/SignUp/index.js

```diff
@@ -35,10 +38,80 @@ export default class SignUpForm extends Component {
         .then(res => this.setState({ school_options: res }));
     }
 
+    measureStrength = (password) => {
+      let score = 0;
+      let passwordStrength;
+      const regexPositive = [
+        '[A-Z]',
+        '[a-z]',
+        '[0-9]',
+        '\\W',
+      ];
+      regexPositive.forEach((regex, index) => {
+        if (new RegExp(regex).test(password)) {
+          score += 1;
+        }
+      });
+      switch (score) {
+        case 0:
+        case 1:
+          passwordStrength = 'weak: ';
+          break;
+        case 2:
+        case 3:
+          passwordStrength = 'good: ';
+          break;
+        case 4:
+        case 5:
+          passwordStrength = 'strong: ';
+          break;
+      }
+      this.setState({
+        passwordStrength,
+        passValidate: 'show-check',
+      });
+    }
+
+    validate = (e) => {
+      const password = e.target.value;
+      let errorMessage;
+      let capsCount; let smallCount; let numberCount; let
+        symbolCount;
+      if (password.length < 8) {
+        this.setState({
+          errorMessage: 'password must be min 8 char',
+          passValidate: 'show-check',
+        });
+      } else {
+        capsCount = (password.match(/[A-Z]/g) || []).length;
+        smallCount = (password.match(/[a-z]/g) || []).length;
+        numberCount = (password.match(/[0-9]/g) || []).length;
+        symbolCount = (password.match(/\W/g) || []).length;
```

Comment:
@donia98 This password strength and validation code is really good.  One suggestion I have (or an idea for next time - it's not an urgently needed change) is to refactor so that the four password regex rules are defined in just one place, for example in a const at the top of this file, instead of appearing in both validate() and measureStrength(), and then both of those functions refer to the one set of rules.  That way, you could add new rules to the list quickly and easily, and both of your functions would use the new set of rules.

Reply:
Thanks, I am agreed with your advice and I will use it in the next time :+1: